signal_scanner_bot/env.py

```diff
@@ -1,20 +1,56 @@
 import logging
 import os
+from pathlib import Path
 from threading import Lock
-from typing import Any, Callable, List, Set
+from typing import Any, Optional, Callable, List, Set
 
 
 log = logging.getLogger(__name__)
 
 
 _VARS = []
+START_LISTENING = "AUTOSCANON"
+STOP_LISTENING = "AUTOSCANOFF"
+START_LISTENING_NOTIFICATION = "==Auto Scanning Activated=="
+STOP_LISTENING_NOTIFICATION = "==Auto Scanning Deactivated=="
 
 
 class _State:
     """Class for holding global state across threads/tasks"""
 
-    LISTENING = False
-    STOP_REQUESTED = False
+    # Initialize state, set LISTENING to true if state file exists,
+    # sets false if not.
+    def __init__(self, file: Path):
+        self.file = file
+        self.LISTENING = self.file.exists()
+        self.STOP_REQUESTED = False
+
+    # Method to update the listening status of the State class
+    # object. Checks for on/off and creates/deletes state file.
+    def update_listening_status(self, status: str) -> str:
+        if status == START_LISTENING:
+            self.LISTENING = True
+            self.file.touch()
+            return START_LISTENING_NOTIFICATION
+        elif status == STOP_LISTENING:
+            self.LISTENING = False
+            self.file.unlink(missing_ok=True)
+            return STOP_LISTENING_NOTIFICATION
+        else:
+            raise ValueError(
+                f"Value {status} provided to _State object, expected AUTOSCANON or AUTOSCANOFF."
+            )
+
+    # Method to return the current state notification message
+    def get_listening_status_notice(self) -> Optional[str]:
+        if self.LISTENING:
+            return START_LISTENING_NOTIFICATION
+        elif not self.LISTENING:
+            return STOP_LISTENING_NOTIFICATION
+        else:
+            # Should be impossible to get her but it's here to
+            # satisfy mypy
+            return None
```

Comment:
Totally your preference, but this logic could alternately be:
```suggestion
        return START_LISTENING_NOTIFICATION if self.LISTENING else STOP_LISTENING_NOTIFICATION
```
That might allow us to change the `Optional[str]` return type to `str` and prevent having to cast it to string when being called.